libsol/parser.c

```diff
@@ -138,24 +134,50 @@ int parse_version(Parser* parser, MessageHeader* header) {
 
 int parse_message_header(Parser* parser, MessageHeader* header) {
     BAIL_IF(parse_version(parser, header));
-    BAIL_IF(parse_pubkeys(parser, &header->pubkeys_header, &header->pubkeys));
+    BAIL_IF(parse_pubkeys_header(parser, &header->pubkeys_header));
+    BAIL_IF(parse_pubkeys(parser, header->pubkeys_header.pubkeys_length, &header->pubkeys));
     BAIL_IF(parse_blockhash(parser, &header->blockhash));
     BAIL_IF(parse_length(parser, &header->instructions_length));
     return 0;
 }
 
+int parse_offchain_message_application_domain(
+    Parser* parser,
+    const OffchainMessageApplicationDomain** app_domain
+) {
+    BAIL_IF(check_buffer_length(parser, OFFCHAIN_MESSAGE_APPLICATION_DOMAIN_LENGTH));
+    *app_domain = (const OffchainMessageApplicationDomain*) parser->buffer;
+    advance(parser, OFFCHAIN_MESSAGE_APPLICATION_DOMAIN_LENGTH);
+    return 0;
+}
+
+/**
+ * Field	    Start offset	        Length (bytes)
+ * Signing Domain	0x00	                    16
+ * Header version	0x10	                    1
+ * Application domain	0x11	                    32
+ * Message format	0x31	                    1
+ * Signer count	        0x32	                    1
+ * Signers	        0x33	                    SIGNER_COUNT * 32
+ * Message length	0x33 + SIGNER_CNT * 32	    2
+ */
```

Comment:
can you get these lined up?

Reply:
What tool are you using for code review? On my end it is formatted quite well (github and CLion), so it's hard to fix because I'm not really able to reproduce it.
Maybe we can just replace it with the link to the docs? This way we don't have to worry about formatting

<img width="568" alt="image" src="https://github.com/blockydevs/app-solana/assets/142581992/d4fa068b-db17-4005-8f16-76986046c9db">

Reply:
I've changed it, I hope that it helps on you end, if not we can remove it and leave just the link